### PR TITLE
Add update() method implementation. Add pop() method implementation

### PR DESCRIPTION
**Pull request by bmwant21**
Add update and pop methods as in original implementation of set

### CODE REVIEW

File: ordered_set.py [hunk 1]
```diff
@@ -112,6 +112,21 @@ def add(self, key):
...
     append = add
 
+    def update(self, sequence):
+        """
+        Update set with given iterable sequence, then return index of last element.
+        If `sequence` is not iterable then just add this element
+        """
+        try:
+            iter(sequence)
+        except TypeError:
```
Comment (rspeer) [inline ordered_set.py]: Why would we need this case? It seems pretty unexpected to me. It doesn't match what `set` does and it seems that it wuold hide bugs.

Passing a non-iterable to `.update()` should be an error.

Comment (bmwant21) [inline ordered_set.py]: Ok, I agree with you so behavior is changes

**Comment by viachk** [comment]
Awesome PR, just needed to implement the `.update()` feature in my project. 
+1 to get it in upstream.

**Comment by rspeer** [comment]
Okay. I just want to make a couple of stylistic changes, so I'll pull it and then make those changes.

**Pull request merged**